apps/wallet/src/ui/app/shared/faucet/useFaucetMutation.ts

```diff
@@ -26,15 +26,47 @@ export function useFaucetMutation(options?: UseFaucetMutationOptions) {
 				throw new Error('Failed, faucet host not found.');
 			}
 
-			const { error, transferredGasObjects } = await requestSuiFromFaucetV0({
+			const { error, task: taskId } = await requestSuiFromFaucetV1({
 				recipient: addressToTopUp,
 				host: options.host,
 			});
+			if (!taskId) {
+				throw new Error('Failed, task id not found.');
+			}
+			// Initialize a variable to track possible faucet request status errors
+			let faucetStatusError: string | null = null;
+			let totalMistTransferred: number | null = null;
+
+			// Continuously check the status until it's no longer 'INPROGRESS'
+			let currentStatus = 'INPROGRESS';
+			while (currentStatus === 'INPROGRESS') {
+				const { status, error } = await getFaucetRequestStatus({
+					host: options.host,
+					taskId,
+				});
+
+				if (status?.status === 'SUCCEEDED') {
+					totalMistTransferred = status.transferred_gas_objects?.sent.reduce(
+						(total, { amount }) => total + amount,
+						0,
+					);
+				}
+
+				if (status?.status !== 'INPROGRESS' || error) {
```

Comment:
If status is `SUCCEDDED` this will still throw

Reply:
I revert back to break. Check the condition below